### DISCUSSION TITLE
content from Component class

### DISCUSSION THREAD

**Discussion by Zucovsky** [KyoriPowered/adventure#967] [Q&A]
hi, how i have a Component class, which returns this
`TextComponentImpl{content="test text", style=StyleImpl{obfuscated=not_set, bold=not_set, strikethrough=not_set, underlined=not_set, italic=not_set, color=null, clickEvent=null, hoverEvent=null, insertion=null, font=null}, children=[]}`
how i can get only content from this class?

**Comment by zml2008** [comment]
What are you trying to do with the `Component` instance? There are very few cases where you'd need the plain text -- and you seem to be `toString()`-ing it for some reason.

**Reply by Zucovsky** [reply]
if I do `toString()`, then I output this string, and I need to get `content`

**Reply by MiniDigger** [reply]
zml asked for the xy, why do you need the content, what do you want to do with that? because depending on what you do, there are multiple possible solutions (calling toString is always wrong tho)
